pvlib/iam.py

```diff
@@ -527,3 +527,120 @@ def sapm(aoi, module, upper=None):
         iam = pd.Series(iam, aoi.index)
 
     return iam
+
+
+def marion_integrate(function, surface_tilt, region, N=180):
```

Comment:
@cwhanse I originally used a keyword to switch between the available IAM functions like you describe, but I switched to passing in a function so that people could use their own model if they wanted.  It seemed useful when I was writing the example, but in hindsight I think I agree it should just be consistent with the rest of pvlib.  

> I would also prefer that the function returned separate IAM modifiers for sky and ground diffuse irradiance.

To be clear, does this mean removing the `region` parameter and calculating all three IAM values instead of just one?

Reply:
> To be clear, does this mean removing the region parameter and calculating all three IAM values instead of just one?

Yes, that's what I meant to say.

Reply:
On second thought the `marion_integrate` function has merit, maybe add `marion_diffuse` as a new function that accepts a kwarg `model` and uses `marion_integrate` to do the calculation. I think the `model` kwarg` would be more familiar to many users than passing a function handle as an argument.